### PR TITLE
fix: contributorIds field is not displayed correctly in metadata view

### LINKED ISSUE #10: 'contributorId' field not displaying correctly in Metadata view
## What happened?

When I look at an author under the Metadata view, the 'contributorId' field is not displaying correctly.

## Can it be recreated?

1. Go to https://libero-editor--test.elifesciences.org/?archive=00104#workflow=edit-metadata-workflow
2. Scroll down, and notice the overflowing field showing the raw JSON of the field
Include screenshots here if relevant.

## What did you expect to happen?

Rather than a JSON dump, the 'card' for an author should itemise each property of the JSON and display that as a set of fields, maybe under an appropriately named header.

### PR DESCRIPTION
**Pull request by ilya-zhuravel**
## Why

contributorIds is not displayed correctly in Metadata editing workflow

## What

I created a custom editor and wired it into Person model to have a correct editor for the contributorIds field

Closes #10

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ package-lock.json
 /tmp
 /coverage
 /.nyc_output
+/.idea
```
Comment (NuclearRedeye) [inline .gitignore:18]: Assume this is an artefact of your IDE?  If so, could you group them under a comment just so it's clear? e.g...

# VS Code
/.vs_code

File: src/article/metadata/ContributorIdentifierEditor.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    domHelpers.stopAndPrevent(e);
+  }
+}
+
+class ContributorIDEditorComponent extends DefaultNodeComponent {
+  _getClassNames() {
+    return 'sc-reference sm-metadata';
+  }
+
+  _createPropertyModels() {
```
Comment (NuclearRedeye) [inline src/article/metadata/ContributorIdentifierEditor.js]: Looking at this, do you need it? I don't think you need the 'stringDate' part, hence you can just always return the createValueModel() part, in which case is the default implementation not enough? I.e. if you delete the function does it still work the same?

File: src/article/shared/ManuscriptContentPackage.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ import {
   DefaultNodeComponent,
   ModelPreviewComponent,
 } from '../components';
+import { PersonComponent } from "../components/PersonComponent";
```
Comment (NuclearRedeye) [inline src/article/shared/ManuscriptContentPackage.js]: If you export this in ../components/index.js, you could add this to the above import statement?

File: src/kit/ui/BooleanComponent.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ export default class BooleanComponent extends ValueComponent {
     if (!this.context.editable) {
       el.addclass('sm-readonly');
     }
-    el.append($$(CheckboxInput, { value }));
+    el.append($$(CheckboxInput, { value, disabled: this.props.disabled }));
```
Comment (NuclearRedeye) [inline src/kit/ui/BooleanComponent.js:18]: Do you need to make sure that the default for disabled is false? Is is this already handled?

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Just a couple of minor tweaks and checks I think

File: src/kit/ui/BooleanComponent.js (hunk 4 above)
Comment (ilya-zhuravel) [reply src/kit/ui/BooleanComponent.js:18]: In this case everything that is falsy will work. We could cast it to boolean explicitly to enforce the type.

File: src/article/metadata/ContributorIdentifierEditor.js (hunk 2 above)
Comment (ilya-zhuravel) [reply src/article/metadata/ContributorIdentifierEditor.js]: Yeah, sorry. It was meant to be deleted but was left out by mistake

File: src/kit/ui/BooleanComponent.js (hunk 4 above)
Comment (NuclearRedeye) [reply src/kit/ui/BooleanComponent.js:18]: I'm happy as long as it doesn't alter the default behaviour from before.

Comment (ilya-zhuravel) [reply src/kit/ui/BooleanComponent.js:18]: it won't

**Review by NuclearRedeye — APPROVED** [review]

**Pull request merged**